tests/testthat/test-csv.R

```diff
@@ -362,18 +360,60 @@ test_that("remaining_columns_to_read() works", {
   )
 })
 
-test_that("read_sample_csv() reads adaptation step size correctly", {
+test_that("read_cmdstan_csv() reads adaptation step size correctly", {
   skip_on_cran()
 
   csv_files <- test_path("resources", "csv", "model1-2-no-warmup.csv")
 
-  csv_out <- read_sample_csv(csv_files)
+  csv_out <- read_cmdstan_csv(csv_files)
   expect_equal(csv_out$step_size[[2]], 0.672434)
 
   csv_files <- c(test_path("resources", "csv", "model1-1-dense_e_metric.csv"),
                  test_path("resources", "csv", "model1-2-dense_e_metric.csv"))
 
-  csv_out <- read_sample_csv(csv_files)
+  csv_out <- read_cmdstan_csv(csv_files)
   expect_equal(csv_out$step_size[[1]], 0.11757)
   expect_equal(csv_out$step_size[[2]], 0.232778)
 })
+
+test_that("read_cmdstan_csv() works for optimize", {
+  skip_on_cran()
+
+  csv_output_1 <- read_cmdstan_csv(fit_bernoulli_optimize$output_files())
+  csv_output_2 <- read_cmdstan_csv(fit_logistic_optimize$output_files())
+  expect_equal(dim(csv_output_1$point_estimates), c(1, 2))
+  expect_equal(dim(csv_output_2$point_estimates), c(1, 5))
+
+  csv_file <- test_path("resources", "csv", "bernoulli-1-optimize.csv")
+  csv_output_3 <- read_cmdstan_csv(csv_file)
+  expect_equal(as.numeric(csv_output_3$point_estimates[1,"lp__"]), -12.2173)
+  expect_equal(as.numeric(csv_output_3$point_estimates[1,"theta"]), 0.300001)
+
+  # variable filtering
+  csv_output_4 <- read_cmdstan_csv(fit_logistic_optimize$output_files(), variables = "beta")
+  expect_equal(posterior::variables(csv_output_4$point_estimates), c("beta[1]", "beta[2]", "beta[3]"))
+  csv_output_5 <- read_cmdstan_csv(fit_logistic_optimize$output_files(), variables = c("alpha", "lp__", "beta[2]"))
+  expect_equal(posterior::variables(csv_output_5$point_estimates), c("alpha", "lp__", "beta[2]"))
+})
```

Comment:
I added these tests to make sure filtering works for optimization. It was causing an error before. Same with variational.